apps/smart-contracts/token/test/Vesting.test.ts

```diff
@@ -1054,4 +1129,132 @@ describe('Vesting', () => {
+      await mineBlock(ethers.provider as any, timeAfterVestingStarted)
+      expect(await vesting.getClaimableAmount(blockedUser.address)).to.be.gt(0)
+
+      await expect(vesting.connect(blockedUser).claim()).revertedWith('Recipient blocked')
+    })
+
+    it('claims if user not blocked', async () => {
+      expect(await blocklist.isIncluded(unblockedUser1.address)).to.eq(false)
+      const unblockedUserBalanceBefore = await ppo.balanceOf(unblockedUser1.address)
+      const vestingBalanceBefore = await ppo.balanceOf(vesting.address)
+      const unblockedUserTotalClaimedBefore = await vesting.getClaimedAmount(unblockedUser1.address)
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      await utils.setNextTimestamp(ethers.provider as any, timeAfterVestingStarted)
+
+      await vesting.connect(unblockedUser1).claim()
+
+      const unblockedUserTotalClaimedAfter = await vesting.getClaimedAmount(unblockedUser1.address)
+      const expectedChangeInBalance = unblockedUserTotalClaimedAfter.sub(
+        unblockedUserTotalClaimedBefore
+      )
+      expect(expectedChangeInBalance).to.not.eq(0)
+      expect(await ppo.balanceOf(unblockedUser1.address)).to.eq(
+        unblockedUserBalanceBefore.add(expectedChangeInBalance)
+      )
+      expect(await ppo.balanceOf(vesting.address)).to.eq(
+        vestingBalanceBefore.sub(expectedChangeInBalance)
+      )
+    })
+
+    describe('if user claims, but is subsequently blocked', () => {
+      beforeEach(async () => {
+        await blocklist.connect(owner).set([blockedUser.address], [false])
+        expect(await blocklist.isIncluded(blockedUser.address)).to.eq(false)
+        const blockedUserBalanceBefore = await ppo.balanceOf(blockedUser.address)
+        const vestingBalanceBefore = await ppo.balanceOf(vesting.address)
+        const blockedUserTotalClaimedBefore = await vesting.getClaimedAmount(blockedUser.address)
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        await utils.setNextTimestamp(ethers.provider as any, timeAfterVestingStarted)
+        // First claim by claimer
+        await vesting.connect(blockedUser).claim()
+        const blockedUserTotalClaimedAfter = await vesting.getClaimedAmount(blockedUser.address)
+        const expectedChangeInBalance = blockedUserTotalClaimedAfter.sub(
+          blockedUserTotalClaimedBefore
+        )
+        expect(expectedChangeInBalance).to.not.eq(0)
+        expect(await ppo.balanceOf(blockedUser.address)).to.eq(
+          blockedUserBalanceBefore.add(expectedChangeInBalance)
+        )
+        expect(await ppo.balanceOf(vesting.address)).to.eq(
+          vestingBalanceBefore.sub(expectedChangeInBalance)
+        )
+        await blocklist.connect(owner).set([blockedUser.address], [true])
+        expect(await blocklist.isIncluded(blockedUser.address)).to.eq(true)
+      })
+
+      it('reverts if tries claiming again', async () => {
```

Comment:
```suggestion
      it('reverts if tries claiming again', async () => {
```
Lets move this at the end of the block to be consistent with other tests (the tests in PR 294 and 295 have their second claiming attempts at the end).